### PR TITLE
fix: buggy ptr in`get_tx_info` syscall inside `validate_declare` for account contracts

### PR DESCRIPTION
**Pull request by whichqua**
Issue Number: #437

## Type

- [ ] feature
- [x] bugfix
- [ ] dev (no functional changes, no API changes)
- [ ] fmt (formatting, renaming)
- [ ] build
- [ ] docs
- [ ] testing

## Breaking changes?

- [ ] yes
- [x] no

### CODE REVIEW

**Review by GMKrieger — APPROVED** [review]
Great work!

**Review by ftheirs — APPROVED** [review]
This looks great! 😎

File: tests/integration/contracts/blockifier_contracts/feature_contracts/cairo0/account_with_syscall_checks.cairo [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    get_tx_info,
+    get_tx_signature,
+    library_call,
+    library_call_l1_handler,
+    replace_class,
+    storage_read,
+    storage_write,
+)
+ 
+func check_all{syscall_ptr: felt*}() {
```
Comment (notlesh) [inline tests/integration/contracts/blockifier_contracts/feature_contracts/cairo0/account_with_syscall_checks.cairo:24]: A short description here would help

**Review by notlesh — APPROVED** [review]
Looks good! `ptr` strikes again...

File: tests/integration/contracts/blockifier_contracts/feature_contracts/cairo0/account_with_syscall_checks.cairo (hunk 1 above)
Comment (whichqua) [reply tests/integration/contracts/blockifier_contracts/feature_contracts/cairo0/account_with_syscall_checks.cairo:26]: Resolved by https://github.com/keep-starknet-strange/snos/pull/441/commits/dc0c5ed511169ee104bda2c05fe1d376d50eecc5

**Pull request merged**